### ISSUE TITLE
[Bug]: system metric (RSS) seems to account for only main process (pytorch `num_workers` > 0)

### ISSUE THREAD

**Issue by davidgill97** [wandb/wandb#10464]
### Describe the bug

The system metric, Process Memory In Use (MB), seems to measure memory usage of the main process when using pytorch and DataLoader with num_workers > 0. 

Below is the system metric logged on wandb
<img width="604" height="304" alt="Image" src="https://github.com/user-attachments/assets/48688c25-c65d-4767-aa58-92dedd38aa2a" />

But actual usage visible in task manager is
<img width="1164" height="195" alt="Image" src="https://github.com/user-attachments/assets/686ddd57-1cb5-436a-b48a-ce4e218c3749" />
The column "working set (memory)" (6th column) shows that each worker occupies ~3GB, and the main memory occupies 4.318GB, which seems to match the log.

The versions:
- wandb: 0.21.0
- python: 3.11.13
- Windows 11 Pro
